src/utils/database-url.ts

```diff
@@ -0,0 +1,31 @@
+import type { DbConfigInterface } from '../interfaces/project-create-interface';
+
+export default function buildDatabaseUrl(config: DbConfigInterface) {
+  let connectionString;
+
+  if (!config) {
+    return null;
+  }
+
+  if (config.dbConnectionUrl) {
+    connectionString = config.dbConnectionUrl;
+  } else {
+    let protocol = config.dbDialect;
+    let port = `:${config.dbPort}`;
+    let password = '';
+
+    if (config.dbDialect === 'mongodb' && config.mongodbSrv) {
+      protocol = 'mongodb+srv';
+      port = '';
+    }
+
+    if (config.dbPassword) {
+      // NOTICE: Encode password string in case of special chars.
+      password = `:${encodeURIComponent(config.dbPassword)}`;
+    }
+
+    connectionString = `${protocol}://${config.dbUser}${password}@${config.dbHostname}${port}/${config.dbName}`;
```

Comment:
It's less likely, but dbUser, dbHostname and dbName name also contain special chars.

Encoding URLs is natively supported in JS (with the URL class), we should really be using that instead of error-prone string manipulation.

```
const url = new URL('protocol://');
url.hostname = 'xxx'
url.port = String(2343)
url.username = 'romain';
url.password = '234234';
return url.href
```